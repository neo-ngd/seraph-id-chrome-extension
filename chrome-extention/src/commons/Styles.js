import { makeStyles } from '@material-ui/core';

const useStyles = makeStyles(theme => ({
  container: {
    display: 'flex',
    flexDirection: 'column',
  },
}));

export default useStyles;
