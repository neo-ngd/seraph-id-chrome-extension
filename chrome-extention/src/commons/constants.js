// Copyright (c) 2019 Swisscom Blockchain AG
// Licensed under MIT License

export const ENCRYPTED_PW_MSG = 'ENCRYPTED_PW_MSG';
export const IMPORT_SUCCESS_MSG = 'IMPORT_SUCCESS_MSG';
export const IMPORT_ERROR_MSG = 'IMPORT_ERROR_MSG';
export const SHARE_ACCOUNT_MSG = 'SHARE_ACCOUNT_MSG';
export const ADD_CLAIM_SUCCESS_MSG = 'ADD_CLAIM_SUCCESS_MSG';
export const ADD_CLAIM_ERROR_MSG = 'ADD_CLAIM_ERROR_MSG';
export const SHARE_CLAIM_SUCCESS_MSG = 'SHARE_CLAIM_SUCCESS_MSG';
export const SHARE_CLAIM_ERROR_MSG = 'SHARE_ADD_CLAIM_ERROR_MSG';

export const SUCCESS = 'SUCCESS';
export const ERROR = 'ERROR';
export const WALLET_NAME = 'SERAPH_ID_WALLET';

export const DIALOG_TYPES = {
    GET_CLAIM: 'GET_CLAIM',
    ASK_CLAIM: 'ASK_CLAIM',
};

export const EVENT_NAMES = {
    CLAIM_ERROR: 'claimError',
    SHARE_CLAIM_SUCCESS: 'shareClaimSuccess',
    SHARE_CLAIM_ERROR: 'shareClaimError',
    ADD_CLAIM_SUCCESS: 'addClaimSuccess',
    ADD_CLAIM_ERROR: 'addClaimError',
    SHARE_ACCOUNT: 'shareAccount',
    ASK_CLAIM: 'askClaim',
    SEND_CLAIM: 'sendClaim',
    GET_ADDRESS: 'getAddress',
};
