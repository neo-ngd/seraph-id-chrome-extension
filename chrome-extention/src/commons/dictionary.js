// Copyright (c) 2019 Swisscom Blockchain AG
// Licensed under MIT License

export default {
    errors: {
        unknown: 'unknown error',
        invalidPassword: 'Invalid password',
        noCredentials: 'credential doesn\'t exists',
        noWallet: 'cannot find any wallet',
        claimDecline: 'user didn\'t accept to share the credential'
    },
    copyButton: {
        copy: 'Copy address!',
        copied: 'Copied!',
    },
    dialogs: {
        askAccept: 'Do you want to accept this claim?',
        wantTo: 'wants to access your',
        askShare: 'Do you want to share it?'
    },
    commons: {
        yes: 'Yes',
        no: 'No',
        address: 'Address',
        claims: 'Claims',
        password: 'Password',
        account: 'Account',
        accounts: 'Accounts',
        'import': 'import',
        another: 'another',
        'export': 'export',
        wallet: 'wallet',
    },
    accountModal: {
        exportWallet: 'Export wallet',
        importWallet: 'Import wallet',
    },
    createWallet: {
        title: 'Please choose a secure password',
        info: 'The password will be used to lock the extension after you close the browser and to encrypt the wallet if you decide to export it... so don’t forget it!',
        passwordInfo: 'Chose a password of at least 5 characters',
    },
    unlockWallet: {
        title: 'Welcome back',
        error: 'Invalid password. Try again.',
        info: 'To continue, please unlock your wallet',
        unlock: 'Unlock Wallet'
    },
    walletInfo: {
        info: 'No Claims yet? Why don’t you play around with our ',
        demo: 'demo',
    },
    welcome: {
        title: 'Welcome to Seraph ID',
        info: 'Seraph ID chrome extention allows you to manage your claims in a user-frielndly way',
        more: 'To start to use Seraph ID please create a wallet or import an existing one',
        link: 'Or import an existing one',
        button: 'Create a Wallet'
    },
    form: {
        error: 'Something went wrong! Try again.',
        success: 'Your wallet has been imported.',
        info: 'Please select a wallet to import.'
    },
    events: {
        saveClaimSuccess: 'The claim has been saved',
        importWalletSuccess: 'Your wallet has been imported',
    }
}
