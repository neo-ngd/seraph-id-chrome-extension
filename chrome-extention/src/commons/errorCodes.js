export const UNKNOWN = 'err:unknown';
export const NOT_FOUND = 'err:notFound';
export const NO_WALLET = 'err:noWallet';
export const CLAIM_DECLINE = 'err:decline';
export const INVALID_PW = 'err:invalidPW';
