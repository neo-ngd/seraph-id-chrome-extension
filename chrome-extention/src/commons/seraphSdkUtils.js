import { DIDNetwork, SeraphIDWallet, SeraphIDIssuer, SeraphIDVerifier } from '@sbc/seraph-id-sdk';
import env from '../environments/environment';
import { v4 as uuid } from 'uuid';
const { GOVERNMENT_SCRIPT_HASH, NEO_RPC_URL, NEOSCAN_URL } = env;

/**
 * Create a new wallet
 * @param wallet
 * @return {SeraphIDWallet}
 */
export const createWallet = wallet => new SeraphIDWallet(wallet);

/**
 * Decrypt the wallet
 * @param accountFromStore
 * @param password
 * @return {Promise<SeraphIDWallet | null>}
 */
export const decrypt = async (accountFromStore, password) => {
  try {
    const wallet = createWallet(JSON.parse(accountFromStore));
    const accounts = wallet.accounts.map(acc => acc.decrypt(password));
    await Promise.all(accounts);
    return wallet;
  } catch (e) {
    return null;
  }
};

/**
 * Generate new DID for private network
 * @param wallet
 */
export const createDid = wallet => wallet.createDID(DIDNetwork.PrivateNet);

/**
 * Get DID
 * @param wallet
 * @return {string}
 */
export const getDid = wallet => `did:neoid:priv:${wallet.accounts[0].label}`;

/**
 * Create new issuer
 * @return {SeraphIDIssuer}
 */
export const createIssuer = () =>
  new SeraphIDIssuer(GOVERNMENT_SCRIPT_HASH, NEO_RPC_URL, NEOSCAN_URL);

/**
 * Create a claim
 * @param schemaName
 * @param data
 * @param wallet
 * @return {IClaim}
 */
export const createClaim = (schemaName, data, wallet) => {
  const issuer = createIssuer();
  issuer.registerNewSchema(schemaName, Object.keys(data), true);
  return issuer.createClaim(uuid(), schemaName, data, getDid(wallet))
};
