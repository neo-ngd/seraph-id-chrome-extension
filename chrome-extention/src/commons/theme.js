import {createMuiTheme} from "@material-ui/core";

const theme = createMuiTheme({
    palette: {
        text: {
            primary: '#FFF',
            secondary: '#CBCFD4',
            light: '#CBCFD4',
            hint: '#00BF0B',
            error: '#FF6E6E',
        },
        primary: {
            main: '#3C444D',
            dark: '#30363D',
            light: '#FFF',
            contrastText: '#00BF0B',
            error: '#FF6E6E',
        },
    },
});

export default theme;
