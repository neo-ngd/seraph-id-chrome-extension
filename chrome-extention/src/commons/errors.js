import dictionary from "./dictionary";

/**
 * Error message string
 * @type {string}
 */
export const ERROR_MSG = 'ERROR';

/**
 * Error codes
 */
export const UNKNOWN = 'err:unknown';
export const NOT_FOUND = 'err:notFound';
export const NO_WALLET = 'err:noWallet';
export const CLAIM_DECLINE = 'err:decline';
export const INVALID_PW = 'err:invalidPW';

/**
 * Send error message to the active tab with content script.
 * @param error
 */
export const sendErrorToCS =  error => {
    chrome.tabs.query({active: true, currentWindow: true}, (tabs) => {
        chrome.tabs.sendMessage(tabs[0].id, {msg: ERROR_MSG, error});
    });
};

/**
 * Send error to the popup.
 * @param error
 */
export const sendErrorToPopup = error => {
    chrome.runtime.sendMessage({msg: ERROR_MSG, error});
};

/**
 * Unknown error
 * @param error
 * @return {{code: *, message: string, error: *}}
 */
export const unknownError = error => ({
    code: UNKNOWN,
    message: dictionary.errors.unknown,
    error
});

/**
 * Invalid password error
 * @return {{code: *, message: string, error: *}}
 */
export const invalidPwError = () => ({
    code: INVALID_PW,
    message: dictionary.errors.invalidPassword,
    error: new Error(dictionary.errors.invalidPassword)
});

/**
 * Claim not found error
 * @return {{code: *, message: string, error: *}}
 */
export const claimNotFoundError = () => ({
    code: NOT_FOUND,
    message: dictionary.errors.noCredentials,
    error: new Error(dictionary.errors.noCredentials)
});

/**
 * Wallet not found error
 * @return {{code: *, message: string, error: *}}
 */
export const walletNotFoundError = () => ({
    code: NO_WALLET,
    message: dictionary.errors.noWallet,
    error: new Error(dictionary.errors.noWallet)
});

/**
 * Claim decline error
 * @return {{code: *, message: string, error: *}}
 */
export const claimDeclineError = () => ({
    code: CLAIM_DECLINE,
    message: dictionary.errors.claimDecline,
    error: new Error(dictionary.errors.claimDecline)
});
