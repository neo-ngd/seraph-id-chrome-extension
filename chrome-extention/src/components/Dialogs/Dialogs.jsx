import React from 'react';
import PropTypes from 'prop-types';
import { Box } from '@material-ui/core';
import ReactJson from 'react-json-view';
import BaseModal from '../Modals/BaseModal';
import BaseButton from '../Buttons/BaseButton';
import {DIALOG_TYPES} from "../../commons/constants";
import Icon from '../Icon/Icon';
import dictionary from "../../commons/dictionary";

const DialogHeader = () => (
    <Box pl={1.2} pt={0.8}>
      <Icon />
    </Box>
);

const Dialogs = ({ open, handleClose, claim, handleClaim, context, schemaName, verifierName }) => {
  return (
    <BaseModal
      HeaderComponent={DialogHeader}
      isOpen={open}
      onClose={handleClose}
      style={{
        maxWidth: '300px',
      }}
    >
      <Box fontSize="16px" pb={2} textAlign="center" color="text.primary">
        {context === DIALOG_TYPES.GET_CLAIM
          ? dictionary.dialogs.askAccept
          : `${verifierName} ${dictionary.dialogs.wantTo} ${schemaName}. ${dictionary.dialogs.askShare}`
        }
      </Box>
      <Box overflow="auto" >
        <ReactJson
            displayObjectSize={false}
            displayDataTypes={false}
            enableClipboard={false}
            src={claim}
          />
      </Box>

      <Box alignSelf="flex-end" display="flex" pt={2} pb={1}>
        <BaseButton
          handleClick={handleClose}
          small
          text={dictionary.commons.no}
          fullWidth={false}
          reject={true}
        />

        <BaseButton
          small
          handleClick={handleClaim}
          text={dictionary.commons.yes}
          fullWidth={false}
        />
      </Box>
    </BaseModal>
  );
};

Dialogs.propTypes = {
  open: PropTypes.bool.isRequired,
  handleClose: PropTypes.func.isRequired,
  claim: PropTypes.object,
  handleClaim: PropTypes.func,
  context: PropTypes.oneOf([
      DIALOG_TYPES.ASK_CLAIM,
      DIALOG_TYPES.GET_CLAIM
  ]),
  schemaName: PropTypes.string,
  verifierName: PropTypes.string,
};

export { DialogHeader };
export default Dialogs;
