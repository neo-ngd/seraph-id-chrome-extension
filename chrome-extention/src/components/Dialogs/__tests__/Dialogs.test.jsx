import React from 'react';
import { shallow } from 'enzyme';
import Dialogs, { DialogHeader } from "../Dialogs";
import {DIALOG_TYPES} from "../../../commons/constants";

const props = {
    open: true,
    handleClose: jest.fn(),
    claim: {test: 'test'},
    handleClaim: jest.fn(),
    context: DIALOG_TYPES.GET_CLAIM,
    schemaName: 'test',
    verifierName: 'test',
};

describe('<Dialogs />', () => {
    it('should render GET_CLAIM component', () => {
        const wrapper = shallow(<Dialogs {...props} />);
        expect(wrapper).toMatchSnapshot()
    });

    it('should render ASK_CLAIM component', () => {
        const wrapper = shallow(<Dialogs {...props} context={DIALOG_TYPES.ASK_CLAIM} />);
        expect(wrapper).toMatchSnapshot()
    });

    it('should render <DialogHeader> component', () => {
        const wrapper = shallow(<DialogHeader />);
        expect(wrapper).toMatchSnapshot()
    });
});
