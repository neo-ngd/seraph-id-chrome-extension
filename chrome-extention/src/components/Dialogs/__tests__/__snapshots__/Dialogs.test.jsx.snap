// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Dialogs /> should render <DialogHeader> component 1`] = `
<Styled(MuiBox)
  pl={1.2}
  pt={0.8}
>
  <Icon />
</Styled(MuiBox)>
`;

exports[`<Dialogs /> should render ASK_CLAIM component 1`] = `
<BaseModal
  HeaderComponent={[Function]}
  isOpen={true}
  onClose={[MockFunction]}
  style={
    Object {
      "maxWidth": "300px",
    }
  }
>
  <Styled(MuiBox)
    color="text.primary"
    fontSize="16px"
    pb={2}
    textAlign="center"
  >
    test wants to access your test. Do you want to share it?
  </Styled(MuiBox)>
  <Styled(MuiBox)
    overflow="auto"
  >
    <t
      collapseStringsAfterLength={false}
      collapsed={false}
      defaultValue={null}
      displayDataTypes={false}
      displayObjectSize={false}
      enableClipboard={false}
      groupArraysAfterLength={100}
      iconStyle="triangle"
      indentWidth={4}
      name="root"
      onAdd={false}
      onDelete={false}
      onEdit={false}
      onSelect={false}
      shouldCollapse={false}
      sortKeys={false}
      src={
        Object {
          "test": "test",
        }
      }
      style={Object {}}
      theme="rjv-default"
      validationMessage="Validation Error"
    />
  </Styled(MuiBox)>
  <Styled(MuiBox)
    alignSelf="flex-end"
    display="flex"
    pb={1}
    pt={2}
  >
    <BaseButton
      disabled={false}
      fullWidth={false}
      handleClick={[MockFunction]}
      icon={false}
      reject={true}
      small={true}
      testID="base-button"
      text="No"
    />
    <BaseButton
      disabled={false}
      fullWidth={false}
      handleClick={[MockFunction]}
      icon={false}
      reject={false}
      small={true}
      testID="base-button"
      text="Yes"
    />
  </Styled(MuiBox)>
</BaseModal>
`;

exports[`<Dialogs /> should render GET_CLAIM component 1`] = `
<BaseModal
  HeaderComponent={[Function]}
  isOpen={true}
  onClose={[MockFunction]}
  style={
    Object {
      "maxWidth": "300px",
    }
  }
>
  <Styled(MuiBox)
    color="text.primary"
    fontSize="16px"
    pb={2}
    textAlign="center"
  >
    Do you want to accept this claim?
  </Styled(MuiBox)>
  <Styled(MuiBox)
    overflow="auto"
  >
    <t
      collapseStringsAfterLength={false}
      collapsed={false}
      defaultValue={null}
      displayDataTypes={false}
      displayObjectSize={false}
      enableClipboard={false}
      groupArraysAfterLength={100}
      iconStyle="triangle"
      indentWidth={4}
      name="root"
      onAdd={false}
      onDelete={false}
      onEdit={false}
      onSelect={false}
      shouldCollapse={false}
      sortKeys={false}
      src={
        Object {
          "test": "test",
        }
      }
      style={Object {}}
      theme="rjv-default"
      validationMessage="Validation Error"
    />
  </Styled(MuiBox)>
  <Styled(MuiBox)
    alignSelf="flex-end"
    display="flex"
    pb={1}
    pt={2}
  >
    <BaseButton
      disabled={false}
      fullWidth={false}
      handleClick={[MockFunction]}
      icon={false}
      reject={true}
      small={true}
      testID="base-button"
      text="No"
    />
    <BaseButton
      disabled={false}
      fullWidth={false}
      handleClick={[MockFunction]}
      icon={false}
      reject={false}
      small={true}
      testID="base-button"
      text="Yes"
    />
  </Styled(MuiBox)>
</BaseModal>
`;
